python/cudf/cudf/tests/test_buffer.py

```diff
@@ -40,19 +42,45 @@ def test_buffer_from_cuda_iface_contiguous(data):
 @pytest.mark.parametrize("dtype", ["uint8", "int8", "float32", "int32"])
 def test_buffer_from_cuda_iface_dtype(data, dtype):
     data = data.astype(dtype)
-    if dtype not in ("uint8", "int8"):
-        with pytest.raises(
-            TypeError, match="Buffer data must be of uint8 type"
-        ):
-            buf = Buffer(data=data, size=data.size)  # noqa: F841
-
-
-@pytest.mark.parametrize("size", [0, 1, 10, 100, 1000, 10_000])
-def test_buffer_copy(size):
-    data = cp.random.randint(low=0, high=100, size=size, dtype="u1")
-    buf = Buffer(data=data)
-    got = buf.copy()
-    assert got.size == buf.size
-    if size > 0:
-        assert got.ptr != buf.ptr
-    assert_array_equal(cp.asarray(buf), cp.asarray(got))
+    as_device_buffer_like(data)
+
+
+@pytest.mark.parametrize("creator", [Buffer, as_device_buffer_like])
+def test_buffer_creation_from_any(creator: Callable[[object], Buffer]):
+    ary = cp.arange(arr_len)
+    b = creator(ary)
+    assert isinstance(b, DeviceBufferLike)
+    assert ary.__cuda_array_interface__["data"][0] == b.ptr
+    assert ary.nbytes == b.size
+
+    with pytest.raises(
+        ValueError, match="size must be specified when `data` is an integer"
+    ):
+        Buffer(42)
+
+
+@pytest.mark.parametrize(
+    "size,expect", [(10, "10B"), (2**10 + 500, "1.49KiB"), (2**20, "1MiB")]
+)
+def test_buffer_repr(size, expect):
+    ary = cp.arange(size, dtype="uint8")
+    buf = as_device_buffer_like(ary)
+    assert f"size={expect}" in repr(buf)
+
+
+@pytest.mark.parametrize(
+    "idx",
+    [
+        2,
+        -2,
```

Comment:
Try with `-1`, I think it will not work.

Reply:
Added, seems to work?

Reply:
ah, you fixed the getitem while I wrote this comment :)